### DISCUSSION TITLE
azureAssistants // retrieval // file_search

### DISCUSSION THREAD

**Discussion by ervet** [danny-avila/LibreChat#4411] [General]
Hello,

i have a working setup with Azure Assistants and some models. Model Builder works. The only thing i can't get to work is the retreival or file_search. It is only greyed out. 

![image](https://github.com/user-attachments/assets/808e9ef5-9e97-4d71-b26c-5d20f451ce5a)

endpoints:
  azureAssistants:
    disableBuilder: false
    pollIntervalMs: 3000
    timeoutMs: 180000
    capabilities: ["code_interpreter", "image_vision", "retrieval", "actions", "tools"]
  azureOpenAI:
    titleModel: "gpt3.5-turbo"
    assistants: true
    titleConvo: true
    plugins: true
    groups:
    - group: "1234"
      apiKey: "1234"
      instanceName: "1234"
      assistants: true
      version: "2024-05-01-preview"
      models:
        gpt4o:
          deploymentName: "gpt-4o"
        gpt3.5-turbo:
          deploymentName: "gpt-35-turbo"
        gpt4:
          deploymentName: "gpt-4"_

I have tried many version and models nothing is working. Any hints - were is this check - on librechat side or Azure side ? 

Greetings
Erik

**Comment by ervet** [comment]
ok .. I got it to work with this parameter

 retrievalModels: ["gpt4o"] 

inside the assistants block, The Name hast to match the model Name not the deployment Name.

**Reply by mansiibm** [reply]
Hey I added this to the yaml as- 

![image](https://github.com/user-attachments/assets/d6adbc43-038c-4bf2-8d8c-8ca4cc32d1b1)

Still the retrieval does not work.

**Comment by leadergxg2020** [comment]
@ervet Could you please post your configuration? thanks

I encountered the same problem.

**Comment by DanielJSimons** [comment]
Hi @ervet ,

Would you kindly be able to provide some details of how this is set up for you? Currently trying and failing to set up retrieval with azure assistants. TIA

**Reply by danny-avila** [reply]
I recommend using Agents, which supports Azure, instead of Azure Assistants: https://www.librechat.ai/docs/features/agents

**Reply by LuHKaeNMS** [reply]
unfortunately the agents are not being accepted /shown in my config once i add them to my yaml file. i already posted a help ;-) any idea ? So You recommend agents instead of assistants ?
